cmd/status-backend/server/server.go

```diff
@@ -53,12 +53,28 @@ func (s *Server) signalHandler(data []byte) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
+	var disconnected []*websocket.Conn
+
 	for connection := range s.connections {
-		err := connection.WriteMessage(websocket.TextMessage, data)
+		err := connection.SetWriteDeadline(time.Now().Add(5 * time.Second))
+		if err != nil {
+			log.Error("failed to set write deadline", "error", err)
+			disconnected = append(disconnected, connection)
+			_ = connection.Close()
+			continue
+		}
+
+		err = connection.WriteMessage(websocket.TextMessage, data)
 		if err != nil {
-			log.Error("failed to write message: %w", err)
+			log.Error("failed to write signal message", "error", err)
+			disconnected = append(disconnected, connection)
+			_ = connection.Close()
```

Comment:
I don't like this duplication, I was thinking of what would be the best solution here. Came up with this. Or do you think it's an overkill? 🤔 

```go
type Server struct {
    mutex sync.RWMutex // note the type change
}

func (s *Server) signalsConnections() map[*websocket.Conn]struct{} {
	s.lock.RLock()
	defer s.lock.RUnlock()

	connections := make(map[*websocket.Conn]struct{}, len(s.connections))
	for conn := range s.connections {
		connections[conn] = struct{}{}
	}
	return connections
}

func (s *Server) dropSignalsConnection(connection *websocket.Conn) {
	s.lock.Lock()
	defer s.lock.Unlock()

	delete(s.connections, connection)
	err := connection.Close()
	if err != nil {
		log.Error("failed to close connection", "error", err)
	}
}

func (s *Server) signalHandler(data []byte) {
	connections := s.signalsConnections()

	for connection := range connections {
		err := connection.SetWriteDeadline(time.Now().Add(5 * time.Second))
		if err != nil {
			log.Error("failed to set write deadline", "error", err)
			s.dropSignalsConnection(connection)
			_ = connection.Close()
			continue
		}

		err = connection.WriteMessage(websocket.TextMessage, data)
		if err != nil {
			log.Error("failed to write signal message", "error", err)
			s.dropSignalsConnection(connection)
		}
	}
}
```

Reply:
Yeah it's fine